### ISSUE TITLE
Add context support to all marketdata.Get* functions

### ISSUE THREAD

**Issue by nathanhack** [alpacahq/alpaca-trade-api-go#308]
All of the functions in the marketdata should accept a context.Context.

Not having it makes the using them prone to long delays waiting for it return before being able to respond to a users requests to close an application.

**Comment by cksidharthan** [comment]
yeah. maybe we can add a `WithContext` for the existing functions so that we dont have breaking changes to the existing library functions.
